src/util/include/util/Interval.h

```diff
@@ -0,0 +1,139 @@
+
+    /**
+     * @brief Computes the convex envelop of this and the other interval
+     * @param other The other interval
+     */
+    [[maybe_unused]] void envelop(const Interval& other) {
+        min = std::min(min, other.min);
+        max = std::max(max, other.max);
+    }
+
+    /**
+     * @brief Computes the convex envelop of this and another value
+     * @param t The other value
+     */
+    [[maybe_unused]] void envelop(const T& t) {
+        if (t >= max) {
+            max = t;
+        } else {
+            min = std::min(min, t);
+        }
+    }
+
+    /**
+     * @brief Test if this interval contains a value
+     * @param t The value
+     * @return true if t lies in *this, false otherwise
+     */
+    [[maybe_unused]] bool contains(const T& t) const { return t > min && t <= max; }
+
+    /**
+     * @brief Test if this interval is contained in another
+     * @param other The other interval
+     * @return true if *this lies inside other, false otherwise
+     */
+    [[maybe_unused]] bool isContainedIn(const Interval& other) const { return min >= other.min && max <= other.max; }
+
+    /**
+     * @brief Test if this interval intersects another
+     * @param other The other interval
+     * @return true if *this and other intersect, false otherwise
+     */
+    [[maybe_unused]] std::optional<Interval<T>> intersect(const Interval& other) const {
+        double newMin = std::max(min, other.min);
+        double newMax = std::min(max, other.max);
+        return (newMin < newMax) ? std::optional(Interval(newMin, newMax)) : std::nullopt;
+    }
+
+    /**
+     * @brief Compute the length of the interval
+     * @return The length
+     */
+    [[maybe_unused]] T length() const { return max - min; }
+
+    bool operator==(const Interval<T>& other) const { return min == other.min && max == other.max; }
+    bool operator!=(const Interval<T>& other) const { return min != other.min || max != other.max; }
+    /**
+     * Lexicographic order for intervals
+     */
+    bool operator<=(const Interval<T>& other) const {
+        return min < other.min || (min == other.min && max <= other.max);
+    }
+    bool operator>=(const Interval<T>& other) const { return other <= *this; }
+    bool operator<(const Interval<T>& other) const { return !(other <= *this); }
+    bool operator>(const Interval<T>& other) const { return !(*this <= other); }
+};
+
+/**
+ * Template class that converts a sorted contiguous container into an iteratable of type Interval
+ * It (virtually) pairs up consecutive elements of the vector 2 by 2 to make Interval's.
+ */
+template <class contiguous_container_type>
+class IntervalIteratable final {
+public:
+    IntervalIteratable(const contiguous_container_type& target): target(target) {}
+    ~IntervalIteratable() = default;
+
+    using value_type = Interval<typename contiguous_container_type::value_type>;
+    using const_iterator = const value_type*;
```

Comment:
It would be cleaner, to Implement the Iterator as a pair of 2 iterators, returning an Interval as copy.

Reply:
Maybe, be unnecessarily less efficient. For a dereferentiation, it'd go from "1 dereferentiation" to "2 dereferentiations + 2 copies into newly allocated mem". I'd rather keep it that way.

Reply:
Don't underestimate the compiler. 
References are often cost free, since they are optimized away. Also you would get rid of the reinterpreted cast, which is extremely dangerous here. Actually this can be UB in the case, that the alignment of the data types are different.
And last but not least, the returned object will be returned by value, with return value optimisation, it will be a noop. And when the object ist used directly, it might not even touch the stack, instead it can stored directly in an register.

Reply:
Ok. I removed the IntervalView class altogether.

Reply:
(Note: we'll still need to have this conversatio for #2729, there are other View's there based on the same principle -- see Spline::SegmentIteratable. I don't really understand your argument yet)